0.57 changelog (#17342)
microsoft/PowerToys: README.md

> - Windows 11 or Windows 10 v1903 (18362) or newer.

Reviewer: ```suggestion
 - Supported Operating Systems:
   - Windows 10 v1903 (18362) or newer.
   - Windows 11 (all versions)
```

This looks better and eliminates the multiple `or` words.

Author: Not a part of the release notes, so I wouldn't touch it for this PR.

- Our installer will install the following items:

>    - [.NET Core 3.1.22 Desktop Runtime](https://dotnet.microsoft.com/download/dotnet/thank-you/runtime-desktop-3.1.22-windows-x64-installer) or a newer 3.1.x runtime. This is needed currently for the Settings application.

Reviewer: ```suggestion
    - [.NET Core 3.1.22 Desktop Runtime](https://dotnet.microsoft.com/download/dotnet/thank-you/runtime-desktop-3.1.22-windows-x64-installer) or a newer 3.1.x runtime. (This is needed currently for the Settings application.)
```

Author: I didn't understand this suggestion, especially since I think all the points need to be on the same level for markdown.

   - [.NET 6.0.3 Desktop Runtime](https://dotnet.microsoft.com/download/dotnet/thank-you/runtime-desktop-6.0.3-windows-x64-installer) or a newer 6.0.x runtime. 

>    - [Microsoft Edge WebView2 Runtime](https://go.microsoft.com/fwlink/p/?LinkId=2124703) bootstrapper. This will install the latest version. 

Reviewer: This part is outdated on the docs.msft page. Should I update this in my docs pr or should we link to this readme (every time up to date)?

Author: That's right, it is. I see no harm in adding it to your PR if you don't mind.

Our [prioritized roadmap][roadmap] of features and utilities that the core team is focusing on.

### 0.57 - March 2022 Update

In this release, we focused heavily on stability and improvements.  Below are some of the highlights!

**Highlights**

- PowerToys Run quality pass. Old standing issues were re-evaluated and fixed.
- Additional features and improvements were added to existing PowerToys Run plugins.

> - New plugin for time and date values/information in PowerToys Run! Thanks [@htcfreek](https://github.com/htcfreek)!

Reviewer: ```suggestion
- New plugin for time and date values/information in PowerToys Run. Thanks [@htcfreek](https://github.com/htcfreek)!
```

A dot is enough here. 😅

Author: Committed.

- The [PowerToys Run documentation](https://aka.ms/PowerToysOverview_PowerToysRun) is also receiving a required update. Thanks [@htcfreek](https://github.com/htcfreek)!

> - We're registering svgs as a Picture kind when SVG Thumbnails are enabled so they appear when searching for pictures in File Explorer.

Reviewer 2: ```suggestion
- We're registering svgs as a picture kind when SVG Thumbnails are enabled so they appear when searching for pictures in File Explorer.
```

> - We're registering svgs as a picture kind when SVG Thumbnails are enabled so they appear when searching for pictures in File Explorer.

Reviewer 3: ```suggestion
- PowerToys will register SVGs as a image file type when SVG Thumbnails are enabled so they appear when searching for images in File Explorer.
```

Reviewer: @jay-o-way
as **an** image file

Author: The name here is actually pictures. and it's a kind. You actually search for "kind:picture" in File explorer.

- We've disabled PDF preview by default, given its incompatibilities with Outlook and that Edge is now being registered for previewing PDF files on Windows 10 too.
- From a coding quality point of view, every project now has code analyzer active. Thanks [@CleanCodeDeveloper](https://github.com/CleanCodeDeveloper)!
- A double click on the tray icon is needed instead of single click to open settings.

### Always on Top

> - New sound for Always on Top activation. Thanks [@franky920920](https://github.com/franky920920)!

Reviewer 3: ```suggestion
- Different sound for Always on Top activation. Thanks [@franky920920](https://github.com/franky920920)!
```

Reviewer: "We changed the sound" makes most sense here.

Author: New gets that novelty feeling ;) It's a new sound being used, even if the sound is not new.

### Awake
- Fixes for the system tray icon. Thanks [@dend](https://github.com/dend)!

> - Temporary duration presets now configurable. Thanks [@dend](https://github.com/dend)!

Reviewer 2: ```suggestion
- Temporary duration presets are now configurable. Thanks [@dend](https://github.com/dend)!
```

- Fix for an issue causing Awake to not be closed properly. Thanks [@dend](https://github.com/dend)!

### ColorPicker

> - It's now possible to delete multiple colors from history and also export them to a file. Thanks [@mshtang](https://github.com/mshtang)!

Reviewer: export the **color values** to file

> - It's now possible to delete multiple colors from history and also export them to a file. Thanks [@mshtang](https://github.com/mshtang)!

Author: I think "a list of colors" also makes sense here.

- The CIEXYZ format has increased precision. Thanks [@m13253](https://github.com/m13253)!
- Performance improved by reducing the use of low level keyboard hooks.

### FancyZones

- Fixed a bug where the same layout applied with different configurations to different screens would reset to a single configuration. (This was a hotfix for 0.56)
- When snapping windows with rounded corners on Windows 11, set the correct corner preferences to avoid gaps between zones. Thanks [@hallatore](https://github.com/hallatore)!
- Fix for canvas layout resetting due to resolution changes.

### File explorer

- Additional markdown file extensions added for Markdown Preview. Thanks [@skycommand](https://github.com/skycommand)!

> - Svgs are now registered as a picture kind on Windows.

Reviewer 3: ```suggestion
- SVG files are now registered as an image file type in Windows.
```

- Added a text wrapping setting and copy context menu to dev file preview. Thanks [@Aaron-Junker](https://github.com/Aaron-Junker)!

> - Pdf file preview and thumbnails are now disabled by default, due to incompatibility with Outlook and an appropriate warning is shown in Settings.

Reviewer 3: ```suggestion
- PDF file previews and thumbnails are now disabled by default, due to incompatibility with Outlook and an appropriate warning is shown in Settings.
```

Author: In the settings, it's "preview"

### Mouse utility

- Find my mouse has a new setting to specify a minimum distance for activation. (This was a hotfix for 0.56)
- Fix for the bug causing the task bar to be hidden behind other windows when a mouse utility was active.
- Fix for the bug causing shortcuts set on icons to not activate when a mouse utility was active.
- Fixed a slight offset in Mouse Pointer Crosshairs when it's configured with an odd thickness.

### PowerToys Run

- New plugin for time and date values/information. Thanks [@htcfreek](https://github.com/htcfreek)!
- WindowWalker now has a tooltip, additional features like killing the process and closing the window, additional settings and improvements. Thanks [@htcfreek](https://github.com/htcfreek)!
- Unit converter now accepts additional ways of writing feet and gallon. Also interprets as either imperial or US gallons depending on current culture. Thanks [@yifan-k](https://github.com/yifan-k)!
- Unit converter now accepts metre for meter.

> - Localization for Web Search and Unit Converter has been enabled.

Reviewer: Let's not mention unit converter here to avoid confusion. The units still not localized.

Author: I'll add a mention it doesn't include units.

Author: Thanks for the review. I've added the mention. PTAL

> - Localization for Web Search and Unit Converter(not including units) has been enabled.

Reviewer: ```suggestion
- Localization for Web Search and Unit Converter (not including units) has been enabled.
```

- Calculator now tries to always interpret the dot (.) symbol as a decimal separator, despite configured culture, to meet expectations.
- Calculator now handles trailing zeroes on hexadecimal numbers correctly.
- System commands plugin can now show the local ip and mac addresses. Thanks [@htcfreek](https://github.com/htcfreek)!
- Folder plugin has improved results, with improved tooltips. Thanks [@htcfreek](https://github.com/htcfreek)!
- Windows settings plugin has added entries for Screen Saver and Connect Wiring Display Panel. Thanks [@htcfreek](https://github.com/htcfreek)!
- Plugins can now show descriptions for their configurations in settings. Thanks [@htcfreek](https://github.com/htcfreek)!
- Fix for the focus issue when calling PowerToys Run for the first time after login and after returning from some windows.
- Fix for a bug on Program when creating a shortcut.
- Validated that upgrading to .net framework 6 fixed the error appearing when shutting down the system with PowerToys Run running.

### Video conference mute

- Newly added microphones are now updated and tracked by VCM.

### Settings

> - What's new button in the bottom with a new look, with a couple more UI tweaks. Thanks [@niels9001](https://github.com/niels9001)!

Reviewer 3: ```suggestion
- _What's new_ button in the bottom with a new look, with a few more UI tweaks. Thanks [@niels9001](https://github.com/niels9001)!
```

- Fixed a bug causing Settings not to open when a racing condition caused Keyboard Manager settings to not be read correctly.
- To open settings from the tray icon a double click is needed instead of a single click.
- Fix for a bug which would cause checking for updates to run indefinitely.

> - When auto-updating, pass a flag to avoid restarting without being prompted.

Reviewer 3: Restarting PT or computer?

Reviewer: good catch. this is about rebooting the computer

### Installer

> - Dependencies installers are now executed with /norestart to avoid unprompted restarts. (This was a hotfix for 0.56). Thanks [@franky920920](https://github.com/franky920920)!

Author: reboots 👍

> - Upgraded .net framework dependency to 6.0.3.

Reviewer 3: ```suggestion
- Upgraded .NET framework dependency to 6.0.3.
```

- Installer logs are now saved where they can be collected and sent by the bug report tool.
- Reverted changes to start with proper elevation and when installed under a different user since those changes ended up causing more issues where PowerToys would start running with the wrong user.

### Development

- OOBE code refactor to have all module information in XAML, like in Settings. Thanks [@niels9001](https://github.com/niels9001)!
- Every project now has analyzers turned on and warnings fixed. Thanks [@CleanCodeDeveloper](https://github.com/CleanCodeDeveloper)!
- New patterns added for code spell-checking and stale entries removed. Thanks [@jsoref](https://github.com/jsoref)
- Additional logging has been added to Fancy Zones and PowerToys Run.
- A new CI release build will not be triggered if all that was changed was just documentation.

> - Fixed a racing condition causing flaky build errors when building Power Rename.

Reviewer 2: ```suggestion
- Fixed a racing condition causing flaky build errors when building PowerRename.
```

Author: You're right, no spaces. Thanks for the suggestions. I've committed them.

- Fixes for the system tray icon. Thanks [@dend](https://github.com/dend)!
- Temporary duration presets now configurable. Thanks [@dend](https://github.com/dend)!
- Fix for an issue causing Awake to not be closed properly. Thanks [@dend](https://github.com/dend)!

### ColorPicker

- It's now possible to delete multiple colors from history and also export them to a file. Thanks [@mshtang](https://github.com/mshtang)!
- The CIEXYZ format has increased precision. Thanks [@m13253](https://github.com/m13253)!
- Performance improved by reducing the use of low level keyboard hooks.

### FancyZones

- Fixed a bug where the same layout applied with different configurations to different screens would reset to a single configuration. (This was a hotfix for 0.56)
- When snapping windows with rounded corners on Windows 11, set the correct corner preferences to avoid gaps between zones. Thanks [@hallatore](https://github.com/hallatore)!
- Fix for canvas layout resetting due to resolution changes.

### File explorer

- Additional markdown file extensions added for Markdown Preview. Thanks [@skycommand](https://github.com/skycommand)!
- Svgs are now registered as a picture kind on Windows.

> - Added a text wrapping setting to dev file preview. Thanks [@Aaron-Junker](https://github.com/Aaron-Junker)!

Reviewer 4: And copy context menu!

Author: Adding a mention to it.

### PowerToys Run

- New TimeDate plugin. Thanks [@htcfreek](https://github.com/htcfreek)!
- WindowWalker has more information in the tooltip, additional features like killing the process and closing the window, additional settings and improvements. Thanks [@htcfreek](https://github.com/htcfreek)!
- Unit converter now accepts additional ways of writing feet and gallon. Also interprets as either imperial or US gallons depending on current culture. Thanks [@yifan-k](https://github.com/yifan-k)!
- Unit converter now accepts metre for meter.
- Localization for Web Search and Unit Converter has been enabled.
- Localization for Windows Terminal has been enabled. (This was a hotfix for 0.56)
- Calculator now tries to always interpret the dot (.) symbol as a decimal separator, despite configured culture, to meet expectations.
- Calculator now handles trailing zeroes on hexadecimal numbers correctly.
- System can now get the local ip and mac addresses. Thanks [@htcfreek](https://github.com/htcfreek)!
- Folder has improved results, with improved tooltips. Thanks [@htcfreek](https://github.com/htcfreek)!

> - Setting has added entries for Screen Saver and Connect Wiring Display Panel. Thanks [@htcfreek](https://github.com/htcfreek)!

Reviewer: ```suggestion
- Windows setting plugin has added entries for Screen Saver and Connect Wiring Display Panel. Thanks [@htcfreek](https://github.com/htcfreek)!
```

- Plugins can now show descriptions for their configurations in settings. Thanks [@htcfreek](https://github.com/htcfreek)!
- Fix for the focus issue when calling PowerToys Run for the first time after login and after returning from some windows.
- Fix for a bug on Program when creating a shortcut.
- Validated that upgrading to .net framework 6 fixed the error appearing when shutting down the system with PowerToys Run running.

### Video conference mute

- Newly added microphones are now updated and tracked by VCM.

### Settings

- What's new button in the bottom with a new look, with a couple more UI tweaks. Thanks [@niels9001](https://github.com/niels9001)!
- Fixed a bug causing Settings not to open when a racing condition caused Keyboard Manager settings to not be read correctly.
- To open settings from the tray icon a double click is needed instead of a single click.
- Fix for a bug which would cause checking for updates to run indefinitely.
- When auto-updating, pass a flag to avoid restarting without being prompted.

### Installer

> - Dependencies installers are now called with /norestart to avoid unprompted restarts. (This was a hotfix for 0.56). Thanks [@franky920920](https://github.com/franky920920)!

Reviewer: ```suggestion
- Dependencies installers are now executed with /norestart to avoid unprompted restarts. (This was a hotfix for 0.56). Thanks [@franky920920](https://github.com/franky920920)!
```

We'd like to directly mention certain contributors (in alphabetical order) for their continued community support this month and helping directly make PowerToys a better piece of software.  
[@Aaron-Junker](https://github.com/Aaron-Junker), [@CleanCodeDeveloper](https://github.com/CleanCodeDeveloper), [@dend](https://github.com/dend), [@franky920920](https://github.com/franky920920), [@htcfreek](https://github.com/htcfreek), [@jay-o-way](https://github.com/jay-o-way), [@jsoref](https://github.com/jsoref), [@niels9001](https://github.com/niels9001), [@snickler](https://github.com/snickler).

#### What is being planned for v0.57
#### What is being planned for v0.58

For [v0.57][github-next-release-work], we'll start work on below:
For [v0.58][github-next-release-work], we'll start work on below:

- Start work on two new PowerToys
- Improvements to PowerToy Run plugins
- Environment Variables Editor PowerToy
- Continue work on another new PowerToy
- Stability / bug fixes
- Validation pass again using WinUI 3.1 for Settings

> - Adding new file types to dev file preview

Reviewer: Can be readded as this is under development.

cc: @Aaron-Junker

Reviewer 4: You're right

Author: Re-adding the new file types to dev file preview
